noir-projects/aztec-nr/authwit/src/account.nr

```diff
@@ -54,6 +56,17 @@ impl AccountActions {
         )
     }
 
+    pub fn initialize_account_contract(self) {
+        let valid_fn = self.is_valid_impl;
+        let mut private_context = self.context.private.unwrap();
+
+        let fee_payload = FeePayload::deserialize(pop_capsule());
+        let fee_hash = fee_payload.hash();
```

Comment:
I think I'm starting to agree with Joe on this use case for capsules. The `fee_payload` should be an argument to this function and not something that shows up magically, and what we really need is a way to say "this argument is not part of the initialization hash". But we can change that in a future PR.

Reply:
I think the tasks for this would be:
- Add a `noinitargcheck` attribute in Noir for the argument to be skipped
- Make sure the attribute is emitted in the abi
- Compute a separate args hash for initialization preimage check if there are any noinitargcheck
- Change the address computation so it ignores args marked as `noinitargcheck` when computing the initialization hash in the address preimage (this will be the most annoying bit I think)
- Ensure the deploy method does _not_ ignore them when actually executing the initialization

It's quite a lot unfortunately, but feels cleaner..?